### ISSUE TITLE
feat: configurable daemon idle timeout (or disable entirely)

### ISSUE THREAD

**Issue by jbrjake** [jbrjake/sahjhan#24]
## Problem

During a Holtz audit, the `awaiting_clear` state requires the user to `/clear` their conversation context and send a new message. The `UserPromptSubmit` hook (`primer.py`) then records a `context_reset` authenticated event via the daemon socket. This is the **only** mechanism that can advance the state machine past `awaiting_clear`.

The problem: during `awaiting_clear`, **no hooks fire**. There's no tool use, no `PreToolUse`/`PostToolUse`, nothing keeping the daemon alive. The user may be away for minutes or hours. If the daemon dies during this window, the session key is gone and the ledger becomes permanently unwritable. The audit is irrecoverably dead.

The daemon currently has no configurable idle timeout. Whatever is causing the daemon to die during idle periods (built-in timeout, OS cleanup, or crash), the result is the same: key loss = audit-terminal.

### 1. `--idle-timeout` flag on `daemon start`

```
sahjhan daemon start --idle-timeout 0      # never timeout (default during audits)
sahjhan daemon start --idle-timeout 3600    # timeout after 1 hour of no socket activity
sahjhan daemon start --idle-timeout 86400   # timeout after 24 hours
```

`0` means "run until explicitly stopped or killed." This should be the default — callers that want a timeout opt in.

### 2. Expose idle timeout in `daemon status`

The `daemon status` response should include the configured idle timeout and time since last socket activity, so callers can monitor:

```json
{"ok": true, "pid": 12345, "uptime_seconds": 3600, "idle_seconds": 120, "idle_timeout": 0, "vault_entries": 2}
```

### 3. Clean shutdown on timeout (not crash)

If the daemon does idle-timeout, it should perform the same clean shutdown as `daemon stop`: zero secrets, remove socket, remove PID file. This way callers can distinguish "daemon was stopped cleanly" from "daemon crashed" by checking for stale files.

## Context

- Holtz issue jbrjake/holtz#37 documents the operational failure: daemon dies during `awaiting_clear`, `context_reset` event never recorded, state machine stuck permanently.
- The daemon's security model (key only in memory, never on disk) means daemon death is **not recoverable**. A restarted daemon generates a new key — the old ledger's HMAC chain is broken. Downstream consumers (Holtz) need to treat daemon death as audit-terminal, but the daemon should also avoid dying unnecessarily.
- The `awaiting_clear` idle window is unbounded — users may step away for hours. No hooks fire during this window to provide keepalive activity. The daemon must survive this.

## Non-goals

- This is not about daemon restart or recovery. Once the key is gone, it's gone.
- This is not about supervision/watchdog. The daemon should simply not die when it doesn't need to.
